### PR TITLE
NEDS-123: updates to packaging for configuring access-point parameter…

### PR DESCRIPTION
**Pull request by bertvi**
Updates to packaging for configuring access-point parameters from env variables

### CODE REVIEW

File: packaging/ap/ubuntu/generic/config [hunk 1]
```diff
@@ -8,55 +8,80 @@ get_prop() {
...
+  if [[ -z "${TOMCAT_PORT}" ]]; then
+    db_input high harmony-ap/tomcatport || true
+  else 
+    db_set harmony-ap/tomcatport "$TOMCAT_PORT"
+  fi
   db_endblock
   db_go
 
   db_beginblock
-  DB_HOST="$(get_prop domibus.database.serverName)"
```
Comment (jhyoty) [inline packaging/ap/ubuntu/generic/config:31]: This is a regression. It is important that the configuration file can be used to override the configuration database value.

File: packaging/ap/ubuntu/generic/config [hunk 2]
```diff
@@ -8,55 +8,80 @@ get_prop() {
...
-  if [[ -n "$DB_HOST" ]]; then
+  if [[ -z "${DB_HOST}" ]]; then
+    db_input high harmony-ap/database-host
+  else 
     db_set harmony-ap/database-host "$DB_HOST"
   fi
-  db_input high harmony-ap/database-host
-  DB_PORT="$(get_prop domibus.database.port)"
```
Comment (jhyoty) [inline packaging/ap/ubuntu/generic/config:36]: This is a regression. It is important that the configuration file can be used to override the configuration database value.

File: packaging/ap/ubuntu/generic/config [hunk 3]
```diff
@@ -8,55 +8,80 @@ get_prop() {
...
-  if [[ -n "$DB_PORT" ]]; then
+  if [[ -z "${DB_PORT}" ]]; then
+    db_input high harmony-ap/database-port
+  else 
     db_set harmony-ap/database-port "$DB_PORT"
   fi
-  db_input high harmony-ap/database-port
-  DB_SCHEMA="$(get_prop domibus.database.schema)"
```
Comment (jhyoty) [inline packaging/ap/ubuntu/generic/config:41]: This is a regression. It is important that the configuration file can be used to override the configuration database value.

File: packaging/ap/ubuntu/generic/config [hunk 4]
```diff
@@ -8,55 +8,80 @@ get_prop() {
...
   fi
-  db_input high harmony-ap/database-schema
-  db_endblock
-  db_go
-
-  db_get harmony-ap/database-schema
-  DB_SCHEMA="$RET"
-
-  db_beginblock
-  DB_USER="$(get_prop domibus.database.user)"
```
Comment (jhyoty) [inline packaging/ap/ubuntu/generic/config:53]: This is a regression. It is important that the configuration file can be used to override the configuration database value.

File: packaging/ap/ubuntu/generic/config [hunk 5]
```diff
@@ -8,55 +8,80 @@ get_prop() {
...
+  else 
     db_set harmony-ap/database-user "$DB_USER"
   fi
+  if [[ -z "${DB_PASS}" ]]; then
+    db_input high harmony-ap/database-password
+  else 
+    db_set harmony-ap/database-password "$DB_PASS"
+  fi
+  
   db_subst harmony-ap/database-user db_schema "$DB_SCHEMA"
```
Comment (jhyoty) [inline packaging/ap/ubuntu/generic/config:84]: db_subst should be before asking for database user

**Review by jhyoty — CHANGES_REQUESTED** [review]
An more idiomatic alternative for using somewhat arbitrary env variables (presumably for some automated install) would be to pre-seed the debconf database using `debconf-set-selections`.

**Comment by bertvi** [comment]
> A more idiomatic alternative for using somewhat arbitrary env variables (presumably for some automated install) would be to pre-seed the debconf database using `debconf-set-selections`.

I deleted all the changes from "packaging/ap/ubuntu/generic/config" and used `debconf-set-selections` in "packaging/ap/build-packages-for-test.sh".

**Review by jhyoty — APPROVED** [review]

**Pull request merged**